edge/cmd/edgecore/app/server.go

```diff
@@ -76,6 +85,23 @@ offering HTTP client capabilities to components of cloud to reach HTTP servers r
 	return cmd
 }
 
+// runningEnvironmentCheck check the environment before edgecore start
+// if Check failed,  return errors
+func runningEnvironmentCheck() error {
+
+	//if kubelet is running, return error
+	if err := osexec.Command("pgrep", "^kubelet$").Run(); err == nil {
+		return errors.New("Error: kubelet is running, kubelet should not running on edge node ")
```

Comment:
The "pgrep" command will lead to linux binding?

Reply:
Yes, it is. But can edge core run other os?

Reply:
Yes, running on other platforms like windows is within our plan. How about this project? https://github.com/mitchellh/go-ps/issues/8  It can choose different platform and provider the same interface.
cc @kevin-wangzefeng @kadisi

Reply:
Looks like https://github.com/mitchellh/go-ps is out of maintenance?
From license perspective, it's fine to use.

Reply:
@fisherxu it's better to use mitchellh/go-ps to check the running process, and I have done for   that, PTAL

Reply:
OK, sounds good :)